### DISCUSSION TITLE
app  | /bin/sh: 1: /app/tmpmain.exe: not found

### DISCUSSION THREAD

**Discussion by raliqala** [air-verse/air#252] [Q&A]
Hello Good day  

I'm facing an issue, am trying to run my go fiber project inside docker with ```air``` but am getting this error  
```uni-blog  | /bin/sh: 1: /app/tmpmain.exe: not found```  

am using 
Windows 11
Docker desktop
golang latest
air  1.27.10
fiber latest

Here is my docker compose and dockerfile
```
# docker-compose up -d --build
version: "3.8"

services:
  app:
    container_name: uni-blog
    image: app-dev
    build:
      context: .
      target: development
    volumes:
      - ./:/app
    ports:
      - 3000:3000

```

```
FROM golang:1.17 as development

RUN apt update && apt upgrade -y && \
  apt install -y git \
  make openssh-client

RUN curl -fLo install.sh https://raw.githubusercontent.com/cosmtrek/air/master/install.sh \
    && chmod +x install.sh && sh install.sh && cp ./bin/air /bin/air

RUN air -v

WORKDIR /app

COPY go.mod go.sum ./

RUN go mod download

COPY . .

EXPOSE 3000

CMD air
```

I also tried installing air following the readME instructions still it gives me this error  
Please help  

Thanks in advance  

![image](https://user-images.githubusercontent.com/30502407/153258131-e9ff8575-4d49-490f-aa47-3801befb218b.png)

![image](https://user-images.githubusercontent.com/30502407/153257671-c4830ff1-5d12-4ce5-b9c3-a47c878fde88.png)

**Comment by xiantang** [comment]
what the air config in your project?

**Reply by raliqala** [reply]
This my air config, i never touched anything in here
```
root = "."
tmp_dir = "tmp"

[build]
  bin = "tmp\\main.exe"
  cmd = "go build -o ./tmp/main.exe ."
  delay = 1000
  exclude_dir = ["assets", "tmp", "vendor"]
  exclude_file = []
  exclude_regex = []
  exclude_unchanged = false
  follow_symlink = false
  full_bin = ""
  include_dir = []
  include_ext = ["go", "tpl", "tmpl", "html"]
  kill_delay = "0s"
  log = "build-errors.log"
  send_interrupt = false
  stop_on_error = true

[color]
  app = ""
  build = "yellow"
  main = "magenta"
  runner = "green"
  watcher = "cyan"

[log]
  time = false

[misc]
  clean_on_exit = false

```

**Reply by raliqala** [reply]
it seems the problem is here ```/app/tmpmain.exe but maybe it should be /app/tmp/main.exe``` but i don't know how to add that / there

**Comment by Muhammad-Ahmad8** [comment]
Did you find the solution to it?

**Reply by raliqala** [reply]
root = "."
testdata_dir = "testdata"
tmp_dir = "tmp"

[build]
  bin = "tmp/main.exe"
  cmd = "go build -o ./tmp/main.exe ."
  delay = 1000
  exclude_dir = ["assets", "tmp", "vendor", "testdata"]
  exclude_file = []
  exclude_regex = ["_test.go"]
  exclude_unchanged = false
  follow_symlink = false
  full_bin = ""
  include_dir = []
  include_ext = ["go", "tpl", "tmpl", "html"]
  kill_delay = "0s"
  log = "build-errors.log"
  send_interrupt = false
  stop_on_error = true

[color]
  app = ""
  build = "yellow"
  main = "magenta"
  runner = "green"
  watcher = "cyan"

[log]
  time = false

[misc]
  clean_on_exit = false

[screen]
  clear_on_rebuild = false

**Reply by raliqala** [reply]
```bin = "tmp/main.exe"```  based on @ColinEge

**Comment by ColinEge** [comment]
Changing the build line in .air.toml from `bin = "tmp\\main.exe"` to `bin = "tmp/main.exe"` shoud fix the issue with the path. I guess it doesn't like the escaped backslash and wants a forward slash instead.
